overlay/default.nix

```diff
@@ -0,0 +1,30 @@
+final: prev: {
+  bun = final.callPackage ./bun { };
+
+  clang-compile = final.callPackage ./clang-compile { };
+
+  dap-cpp = final.callPackage ./dap-cpp { };
+
+  # TODO: rename dap-python for consistency
+  dapPython = final.callPackage ./dapPython {
+    pypkgs = prev.pythonPackages;
+  };
+
+  java-debug = final.callPackage ./java-debug { };
+
+  jdt-language-server = final.callPackage ./jdt-language-server { };
+
+  phpactor = final.callPackage ./phpactor { };
+
+  pip = final.callPackage ./pip { };
+
+  poetry = final.callPackage ./poetry { };
+
+  python-lsp-server = final.callPackage ./python-lsp-server { };
+
+  replbox = final.callPackage ./replbox { };
+
+  stderred = final.callPackage ./stderred {
+    inherit (prev) stderred;
+  };
+}
```

Comment:
The [NixOS wiki](https://nixos.wiki/wiki/Flakes) describes overlay outputs:

>   Overlay, consumed by other flakes

The purpose of the overlays outputs is to provide a way for external consumers to add packages to their nixpkgs set. 

I don't think we should be using them to improve the internal ergonomics of our flake because it is a more confusing flow of how packages in the flake depend on one another and might cause collisions/confusion with what is already in nixpkgs.

Reply:
> The purpose of the overlays outputs is to provide a way for external consumers to add packages to their nixpkgs set.

I can define the overlay but not provide it as an output?

> I don't think we should be using them to improve the internal ergonomics of our flake

I think there's an argument to be made that overlays can be used for internal ergonomics if done well? I admit I didn't particularly define this overlay tastefully, but ergonomics-wise it can help a lot with deduplication/DRY and (I believe) can walk the fine line between ergonomics and verbosity

> might cause collisions/confusion with what is already in nixpkgs.

Maybe I use an internal-only scope? I can bring back the old `replitPackages` scope or something a little clearly more internal like `internalPackages`